crates/swift-integration-tests/build.rs

```diff
@@ -1,8 +1,8 @@
 use std::path::PathBuf;
 
 fn main() {
-    let out_dir = "../../SwiftRustIntegrationTestRunner/Generated";
-    let out_dir = PathBuf::from(out_dir);
+    // TODO: we can use --artifact-dir with OUT_DIR when that stabilizes.
```

Comment:
What do you mean by "with `OUT_DIR`"? Why would we use both?

Why would we want to use either of these things instead of the hard coded string that we're using now.

Let's make it clear what this TODO is trying to achieve. Why is this beneficial?

This is important because 5 years from now someone else that is working on this might not understand what your intentions were here.

We try to be very clear with our TODOs so that anyone can pick them up when they come across them.

---

Also, let's link to the tracking issue so that a maintainer can more easily check if the feature that we're waiting for it available. https://github.com/rust-lang/cargo/issues/6790

Reply:
I thought it was self-evident that hardcoding the output path is brittle (I stumbled over this). Letting `integration-tests.sh` dictate the output directory seems much better. I can add more to the TODO explicating that.

Reply:
Ah, I have a better sense now.

What I have in mind was that, since this path will likely change less than once every 3+ years, flexibility isn't much of a concern.

So, we'd be trading away the ability to do `cargo build` on this crate for the flexibility of overriding a flag that we never need to override.
Similarly, users that are using this script as inspiration for their own codebase won't benefit from the flexibility since they also probably always want to us the same out dir 100% of the time, for years in a row.

If you still disagree then I'm cool with proceeding with your plan here, just add a reason to the TODO since, at least for me, it isn't obvious that it's an improvement.

> I can add more to the TODO explicating that.

Great. That'd be enough for me. Thanks.